Extensions/dnSpy.Debugger/dnSpy.Debugger.DotNet.Mono/Impl/DbgEngineImpl.cs

```diff
@@ -424,13 +424,19 @@ sealed class DelegatePendingMessage : PendingMessage {
 						var cts = new CancellationTokenSource(connectionTimeout - elapsedTime);
 						var asyncConn = VirtualMachineManager.ConnectAsync(endPoint, null, cts.Token);
 						if (!asyncConn.Wait(connectionTimeout - elapsedTime))
-							throw new CouldNotConnectException(GetCouldNotConnectErrorMessage(connectionAddress, connectionPort, filename));
+							throw new CouldNotConnectException(
+								GetCouldNotConnectErrorMessage(connectionAddress, connectionPort, filename));
 						vm = asyncConn.Result;
 						break;
 					}
 					catch (SocketException sex) when (sex.SocketErrorCode == SocketError.ConnectionRefused) {
 						// Retry it in case it takes a while for mono.exe to initialize or if it hasn't started yet
 					}
+					catch (AggregateException aex) when (aex.InnerExceptions.All(ex => ex is SocketException {
+						                                     SocketErrorCode: SocketError.ConnectionRefused
+					                                     })) {
+						// Retry it in case it takes a while
+					}
```

Comment:
Would it not be better to check if `InnerExceptions` contains only one element which is the `SocketException` instead of testing every single one? Is it possible for `InnerExceptions` to contain more than one `SocketException` from the code inside the try?

Reply:
I don't think it's possible but if it really happens, this logic is still most likely correct and you don't have to fix it again. Anyway, I will change it if you insist.